src/Illuminate/Foundation/Console/ServeCommand.php

```diff
@@ -227,24 +227,22 @@ protected function canTryAnotherPort()
     protected function handleProcessOutput()
     {
         return fn ($type, $buffer) => str($buffer)->explode("\n")->each(function ($line) {
+            $requestPort = $this->getRequestPortFromLine($line);
+            $requestDate = $this->getDateFromLine($line);
+
             if (str($line)->contains('Development Server (http')) {
                 $this->components->info("Server running on [http://{$this->host()}:{$this->port()}].");
                 $this->comment('  <fg=yellow;options=bold>Press Ctrl+C to stop the server</>');
 
                 $this->newLine();
-            } elseif (str($line)->contains(' Accepted')) {
-                $requestPort = $this->getRequestPortFromLine($line);
-
+            } elseif (str($line)->contains(' Accepted') && null !== $requestPort && null !== $requestDate) {
                 $this->requestsPool[$requestPort] = [
-                    $this->getDateFromLine($line),
+                    $requestDate,
                     false,
                 ];
-            } elseif (str($line)->contains([' [200]: GET '])) {
-                $requestPort = $this->getRequestPortFromLine($line);
-
+            } elseif (str($line)->contains([' [200]: GET ']) && null !== $requestPort) {
                 $this->requestsPool[$requestPort][1] = trim(explode('[200]: GET', $line)[1]);
-            } elseif (str($line)->contains(' Closing')) {
-                $requestPort = $this->getRequestPortFromLine($line);
+            } elseif (str($line)->contains(' Closing') && null !== $requestPort && isset($this->requestsPool[$requestPort]) && count($this->requestsPool[$requestPort]) == 2) {
```

Comment:
What is this line actually checking? Why the hard-coded 2?

Reply:
**$requestsPool** variable stores the **Start Time** and **filename** of the handled requests. In this line, we are checking if both start time and file name are present in the $requestsPool[$requestPort] variable, hence the hard-coded value 2.